### ISSUE TITLE
[Feature Request] conf.d as proxy to a conf.d centralized  repository

### ISSUE THREAD

**Issue by toni-moreno** [kelseyhightower/confd#486]
Hi to all guys.

I'm looking for some kind of easy centralized configuration system for a lot  (>6000) of performance system agents  (for linux and for windows) 

The problem of this  approach is i'm not the root/administrator of these systems , and I need lot of flexibility on the configuration files.( they could have changes weekly and sometimes daily)

 I have  installed these agents with puppet but we need to have the ability to change often their configuration files on the fly without root access.

As I know, systems as etcd give us a centralized Key/value repository and conf.d give also the way to put these values on arbitrary config files , but sometimes the changes should be done in the templates better than the k/v and as far as I know we can not change templated configuration files if these files are stored on the same server that I can't access as root. 

I would like to Install  only once the conf.d to get all configuration files from a centralized conf.d server that I can Access , control, configure and modify in the same way I do with etcd

There is any plans to add these feature to conf.d in the future?

Thank you very much for your great work

**Comment by bacongobbler** [comment]
One way this could be handled outside the system is through [gitfs](https://github.com/PressLabs/gitfs) or a separate networked filesystem. Is there a certain use case which can't be solved through existing solutions?

Perhaps you might be interested in https://github.com/kelseyhightower/confd/issues/183.

**Comment by okushchenko** [comment]
Closing this in favor of a discussion in #183
